lib/puppet/provider/cisco_ace/nxapi.rb

```diff
@@ -0,0 +1,222 @@
+                                            ACL_BOOL_PROPS)
+
+  def initialize(value={})
+    super(value)
+    afi = @property_hash[:afi]
+    acl_name = @property_hash[:acl_name]
+    seqno    = @property_hash[:seqno]
+    @ace = Cisco::Ace.aces[afi][acl_name][seqno] unless acl_name.nil? && seqno.nil?
+    @property_flush = {}
+  end
+
+  def self.properties_get(afi, acl_name, seqno, instance)
+    debug "Checking acl instance, #{afi} #{acl_name} #{seqno}"
+    current_state = {
+      name:     "#{afi} #{acl_name} #{seqno}",
+      acl_name: acl_name,
+      seqno:    seqno,
+      ensure:   :present,
+      afi:      afi,
+    }
+
+    # Call node_utils getter for each property
+    ACL_NON_BOOL_PROPS.each do |prop|
+      current_state[prop] = instance.send(prop)
+    end
+    ACL_BOOL_PROPS.each do |prop|
+      val = instance.send(prop)
+      if val.nil?
+        current_state[prop] = nil
+      else
+        current_state[prop] = val ? :true : :false
+      end
+    end
+    debug current_state
+    new(current_state)
+  end # self.properties_get
+
+  def self.instances
+    ace_hash = []
+    Cisco::Ace.aces.each do |afi, acl_aces|
+      acl_aces.each do |acl_name, aces|
+        aces.each do |seqno, ace_instance|
+          ace_hash << properties_get(afi, acl_name, seqno, ace_instance)
+        end
+      end
+    end
+    ace_hash
+  end
+
+  def self.prefetch(resources)
+    ace_instances = instances
+    resources.keys.each do |name|
+      provider = ace_instances.find do |ace|
+        resources[name][:seqno] == ace.seqno.to_i &&
+        resources[name][:acl_name] == ace.acl_name.to_s &&
+        resources[name][:afi] == ace.afi.to_s
+      end
+      resources[name].provider = provider unless provider.nil?
+    end
+  end # self.prefetch
+
+  def exists?
+    @property_hash[:ensure] == :present
+  end
+
+  def create
+    @property_flush[:ensure] = :present
+  end
+
+  def destroy
+    @property_flush[:ensure] = :absent
+  end
+
+  def properties_set(new_ace=false)
+    ACL_ALL_PROPS.each do |prop|
+      next unless @resource[prop]
+      if new_ace
+        # Set @property_flush for the current object
+        send("#{prop}=", @resource[prop])
+      end
+      next if @property_flush[prop].nil?
+      # Call the AutoGen setters for the @ace
+      # node_utils object.
+      @ace.send("#{prop}=", @property_flush[prop]) if
+        @ace.respond_to?("#{prop}=")
+    end
+    # Set methods that are not autogenerated follow.
+    ace_set
+  end
+
+  # Custom setters.
+  # The following properties are setters and cannot be handled
+  # by PuppetX::Cisco::AutoGen.mk_puppet_methods.
+  def ace_set
+    return unless @property_flush[:action] ||
+                  @property_flush[:proto] ||
+                  @property_flush[:src_addr] ||
+                  @property_flush[:src_port] ||
+                  @property_flush[:dst_addr] ||
+                  @property_flush[:dst_port]
+    @ace.ace_set(@property_flush)
```

Comment:
`@property_flush` has more in it than just these 6 properties that you care about. Can you simplify this to something like this:

```
  def ace_set
    attrs = {}
    [:action, :proto, :src_addr, :src_port,  :dst_addr, :dst_port].each do |p|
      attrs[p] = @property_flush[p] if @property_flush[p]
    end
    return if attrs.empty?
    @ace.ace_set(attrs)
```